index.html

```diff
@@ -436,113 +445,116 @@
-        delayed even if a callback runs for the whole time period from
-        <var>now</var> to <var>deadline</var>. As such, it should be set to the
-        minimum of: the closest timeout in the [list of active timers] as set
-        via `[setTimeout]` and `[setInterval]`; the scheduled runtime for
-        pending animation callbacks posted via `[requestAnimationFrame]`;
-        pending internal timeouts such as deadlines to start rendering the next
-        frame, process audio or any other internal task the user agent deems
-        important; and a maximum cap of 50ms in the future to ensure
-        responsiveness to unpredictable user input within the threshold of
-        human perception.</p>
+        active at a given time for any given `window`. The idle period can end
+        early if the user agent determines that it is no longer idle. If so,
+        the next idle period cannot start until after <var>deadline</var>.</p>
       </div>
     </section>
     <section>
       <h2>Invoke idle callbacks algorithm</h2>
       <p>The <dfn>invoke idle callbacks algorithm</dfn>:</p>
       <ol>
-        <li>Let <var>docs</var> be the list of [fully active] `Document`
-        objects whose `Window` object's <a>list of runnable idle callbacks</a>
-        is not empty.
-        </li>
+        <li>If the user-agent believes it should end the idle period early due
+        to newly scheduled high-priority work, skip to step 4.
         <li>Let <var>now</var> be the current time.</li>
         <li>If <var>now</var> is less than <var>deadline</var>:
           <ol>
-            <li>Select any <var>document</var> from <var>docs</var>.</li>
-            <li>Pop the top <var>callback</var> from <var>document</var>'s
-            `Window` object's <a>list of runnable idle callbacks</a>.</li>
-            <li>If <var>document</var>'s `Window` object's <a>list of runnable
-            idle callbacks</a> is now empty, remove <var>document</var> from
-            <var>docs</var>. </li>
+            <li>Pop the top <var>callback</var> from <var>window</var>'s
+            <a>list of runnable idle callbacks</a>.</li>
             <li>Let <var>deadlineArg</var> be a new <a>IdleDeadline</a>.
             Set the <a>time</a> associated with <var>deadlineArg</var> to
             <var>deadline</var> and set the `didTimeout` attribute to
             `false`.</li>
             <li>Call <var>callback</var> with <var>deadlineArg</var> as its
             argument. If an uncaught runtime script error occurs, then [report
             the error].</li>
-            <li>If <var>docs</var> is not empty, the user agent SHOULD [queue a
-            task] which performs the steps in the <a>invoke idle callbacks
-            algorithm</a> with <var>deadline</var> as parameter.</li>
+            <li>If <var>window</var>'s <a>list of runnable idle callbacks</a>
+            is not empty, the user agent SHOULD [queue a task] which performs
```

Comment:
Right this was originally to give scope for ending an idle period early, but that is now done in step one. Done.